src/DataProvider/ChainItemDataProvider.php

```diff
@@ -44,7 +44,15 @@ public function getItem(string $resourceClass, $id, string $operationName = null
                     continue;
                 }
 
-                return $dataProvider->getItem($resourceClass, $id, $operationName, $context);
+                $identifier = $id;
+                if (!$dataProvider instanceof DenormalizedIdentifiersAwareItemDataProviderInterface) {
+                    @trigger_error(sprintf('Receiving "$id" as non-array in an item data provider is deprecated, your item data providers must implement "%s".', DenormalizedIdentifiersAwareItemDataProviderInterface::class), E_USER_DEPRECATED);
```

Comment:
Is it mandatory to trigger a deprecation notice? In most case, you'll not use composed identifiers (it's not really reliable), and implementing this new interface looks like an annoyance for most people to handle specific use cases.

I suggest to trigger a deprecation only if `\count($identifier) > 1`. WDYT?

Reply:
@dunglas it is your call, I've no strong opinion on it.
My argument is the change affects all data providers, even if you don't use composite identifiers. In 2.2 you would get `"2"` passed as `$id` and now you will get `['id' => 2]` so in an ideal world, every data provider needs to be updated. Not triggering the deprecation error is ok, it just means that we won't be able to remove this piece of code in the next major version

Reply:
Making it easy for the user to add new data providers matter, so I suggest to trigger the deprecation only if `\count($identifier) > 1`.

Reply:
Ok I'll update acordingly this morning